lms/static/js/progress_page_additional_form.js

```diff
@@ -28,14 +29,27 @@ $("#profile-form-save-action").on("click", function (e) {
             }
         }
 
+        if(termsOfService.attr('checked') === undefined) {
+            if (!$('.terms-of-service-error').length) {
+                $('<div class="terms-of-service-error">This field is required</div>').insertAfter(termsOfService);
+            }
+            return true;
+        } else {
+            $('.terms-of-service-error').remove();
+        }
+
         if(errorCount) {
             form.find(".form-incomplete-error").text("All fields are required");
             $(".progress-form-modal .modal-body").first().animate({scrollTop: 0}, "slow");
             return errorCount;
         }
     }
 
-    if(!checkEmptyFields() || checkEmptyFields() === undefined) {
+    var isCheckEmptyFields = checkEmptyFields();
+
+    if(!isCheckEmptyFields || isCheckEmptyFields === undefined) {
+
+        console.log(ajaxData);
```

Comment:
Do we really need these logs in production?

Reply:
Removed, done.